lib/improver/utilities/mathematical_operations.py

```diff
@@ -0,0 +1,228 @@
+        if self.direction_of_integration not in ["positive", "negative"]:
+            msg = ("The specified direction of integration should be either "
+                   "'positive' or 'negative'. {} was specified.".format(
+                       self.direction_of_integration))
+            raise ValueError(msg)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<Integration: coord_name_to_integrate: {}, '
+                  'start_point: {}, end_point: {}, '
+                  'direction_of_integration: {}>'.format(
+                      self.coord_name_to_integrate, self.start_point,
+                      self.end_point, self.direction_of_integration))
+        return result
+
+    def ensure_monotonic_increase_in_chosen_direction(self, cube):
+        """Ensure that the chosen coordinate is monotonically increasing in
+        the specified direction.
+
+        Args:
+            cube (Iris.cube.Cube):
+                The cube containing the coordinate to check.
+
+        Returns:
+            cube (Iris.cube.Cube):
+                The cube containing a coordinate that is monotonically
+                increasing in the desired direction.
+
+        Raises:
+            ValueError: The chosen coordinate is not monotonic.
+
+        """
+        coord_name = self.coord_name_to_integrate
+        direction = self.direction_of_integration
+        increasing_order = np.all(np.diff(cube.coord(coord_name).points) > 0)
+
+        if increasing_order and direction == "positive":
+            pass
+        elif increasing_order and direction == "negative":
+            cube = sort_coord_in_cube(cube, coord_name, order="descending")
+        elif not increasing_order and direction == "positive":
+            cube = sort_coord_in_cube(cube, coord_name)
+        elif not increasing_order and direction == "negative":
+            pass
+        return cube
+
+    def process(self, cube):
+        """Integrate a specified coordinate. This is calculated by defining the
+        upper and lower bounds for the steps along a chosen coordinate
+        within the cube.
+
+        Integration is performed by firstly defining the stride as the
+        difference between the upper and lower bound. The contribution from
+        the uppermost half of the stride is calculated by multiplying the
+        upper bound value by 0.5 * stride, and the contribution
+        from the lowermost half of the stride is calculated by multiplying the
+        lower bound value by 0.5 * stride. The contribution from the
+        uppermost half of the stride and the bottom half of the stride is
+        summed.
+
+        As the coordinate is progressively integrated, the contribution of
+        each stride is cumulatively summed.
+
+        Args:
+            cube (Iris.cube.Cube):
+                The cube containing the coordinate to check.
```

Comment:
Incorrect? How about "Cube containing data to be integrated"

Reply:
Changed.